scripts/mpi_passive_recv.cc

```diff
@@ -0,0 +1,206 @@
+// A pure C++ mpi to implement passive receive.
+// Compile and run with
+//   mpicxx -o mpi_passive_recv mpi_passive_recv.cc -std=c++11 && bfrun -np 4 mpi_passive_recv
+
+#include <mpi.h>
+#include <stdio.h>
+
+#include <atomic>
+#include <cassert>
+#include <memory>
+#include <queue>
+#include <thread>
+#include <vector>
+
+const int kSendLength = 20;
+const int kTag = 1234567;
+
+#define MPICHECK(cmd)                                                  \
+  do {                                                                 \
```

Comment:
Why use do while here? If it is for avoiding namespace contamination, I guess a pair of brackets should be enough.

Reply:
I don't know. That is what listed in nccl example.